### ISSUE TITLE
spring-auto-restdocs-json-doclet-jdk9 throw java.lang.ClassCastException

### ISSUE THREAD

**Issue by organics2016** [ScaCap/spring-auto-restdocs#350]
env versions

openjdk 11.0.2
spring-auto-restdocs-core  2.0.6
maven-javadoc-plugin  3.0.1
spring-restdocs-mockmvc  2.0.3.RELEASE

err log 
```
[ERROR] java.lang.ClassCastException: class com.sun.tools.javac.tree.DCTree$DCErroneous cannot be cast to class com.sun.source.doctree.BlockTagTree (com.sun.tools.javac.tree.DCTree$DCErroneous and com.sun.source.doctree.BlockTagTree are in module jdk.compiler of loader 'app')
[ERROR] at capital.scalable.restdocs.jsondoclet.MethodDocumentation.lambda$fromMethodDoc$0(MethodDocumentation.java:57)
[ERROR] at java.base/java.lang.Iterable.forEach(Iterable.java:75)
[ERROR] at capital.scalable.restdocs.jsondoclet.MethodDocumentation.lambda$fromMethodDoc$1(MethodDocumentation.java:51)
[ERROR] at java.base/java.util.Optional.ifPresent(Optional.java:183)
[ERROR] at capital.scalable.restdocs.jsondoclet.MethodDocumentation.fromMethodDoc(MethodDocumentation.java:51)
[ERROR] at capital.scalable.restdocs.jsondoclet.ClassDocumentation.addMethod(ClassDocumentation.java:68)
[ERROR] at capital.scalable.restdocs.jsondoclet.ClassDocumentation.lambda$fromClassDoc$0(ClassDocumentation.java:50)
[ERROR] at java.base/java.lang.Iterable.forEach(Iterable.java:75)
[ERROR] at capital.scalable.restdocs.jsondoclet.ClassDocumentation.fromClassDoc(ClassDocumentation.java:45)
[ERROR] at capital.scalable.restdocs.jsondoclet.ExtractDocumentationAsJsonDoclet.lambda$run$1(ExtractDocumentationAsJsonDoclet.java:68)
[ERROR] at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
[ERROR] at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
[ERROR] at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
[ERROR] at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
[ERROR] at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
[ERROR] at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
[ERROR] at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
[ERROR] at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
[ERROR] at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[ERROR] at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
[ERROR] at capital.scalable.restdocs.jsondoclet.ExtractDocumentationAsJsonDoclet.run(ExtractDocumentationAsJsonDoclet.java:63)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Start.parseAndExecute(Start.java:582)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Start.begin(Start.java:431)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Start.begin(Start.java:344)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Main.execute(Main.java:63)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Main.main(Main.java:52)
```

**Comment by organics2016** [comment]
Tested

env versions

openjdk 11.0.2
spring-auto-restdocs-core 2.0.6
maven-javadoc-plugin 3.0.1
spring-restdocs-mockmvc 2.0.3.RELEASE

run maven package 

Ues  spring-auto-restdocs-json-doclet-jdk9  return above error

Use  spring-auto-restdocs-json-doclet  works as expected.

**Comment by jmisur** [comment]
Hi, our master build has no problems with openjdk 11 as you can see [here](https://travis-ci.org/ScaCap/spring-auto-restdocs/jobs/588612192).
Try to checkout latest release [2.0.6](https://github.com/ScaCap/spring-auto-restdocs/releases/tag/v2.0.6), go to samples folder and run `mvn clean package`. This will run all samples projects with whatever jdk is configured in env. 
Additionally you can also point us to your repo.

**Comment by ChristianSch** [comment]
I just had the same problem. The cause I finally found was a docstring like this:

```
/**
 * something ...
 * @throws etc.
 * @
*/
```

So just having the empty `@` thing, where I was distracted from finishing, cost me 2 days of debugging this weird exception.

**Comment by jmisur** [comment]
Thanks for finding the cause! We'll try to fix the code for this scenario.

**Comment by stefanrybacki** [comment]
what is the ETA for 2.0.8 as we need this fix asap?

**Comment by fbenz** [comment]
Asap, all planned PRs are in and I will try to find some time for it in the next few days.

**Comment by fbenz** [comment]
@stefanrybacki We released 2.0.8. I'm sorry for the delay. We had issues updating the signing key for the artifacts.
